GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc

```diff
@@ -807,6 +807,15 @@ bool Pythia8Hadronizer::hadronize()
     nFSRveto += fEmissionVetoHook->getNFSRveto();  
   }
 
+  // fill shower weights
+  // http://home.thep.lu.se/~torbjorn/pythia82html/Variations.html
+  if( fMasterGen->info.nWeights() > 1 ){
```

Comment:
hi @jfernan2 - why protect against >1 here? It seems this will mean users all have to protect against this case downstream.

Reply:
This is the way it has been done in other parts of the code in L753 and L959. Weights are only saved if their size is >0, that's my understanding. What are you proposing instead? Thanks

Reply:
thanks I had missed that (not that I understand it, but consistency is good)